packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/PropertyDeclaration.cs

```diff
@@ -3,8 +3,92 @@
 
 using System;
 using System.Collections.Generic;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
 
 namespace Microsoft.Generator.CSharp
 {
-    public record PropertyDeclaration(FormattableString? Description, MethodSignatureModifiers Modifiers, CSharpType Type, string Name, PropertyBody Body, IReadOnlyDictionary<CSharpType, FormattableString>? Exceptions = null, CSharpType? ExplicitInterface = null);
+    public class PropertyDeclaration
```

Comment:
I am wondering why we have to change records to classes.
Is there anything we cannot get from records but we have in classes?
I think this is impossible because records are just classes with quite a few compiler generated members.

if we decide to change it, we should also consider to add `DebuggerDisplay` on types like this because currently without records, this will just show up as `Microsoft.Generator.CSharp.PropertyDeclaration` which is just useless.
Previously when it is a record, because record has generated `ToString`, we at least got something for its content when debugging

Reply:
I think the goal here is to only have a ctor which takes the InputModelProperty so it can't be a record as we don't want that to be a public property

Reply:
We have kept the additional ctors to allow creating params and properties that don't come from the spec.